### ISSUE TITLE
Trailing data descriptor in zip file not supported

### ISSUE THREAD

**Issue by alochmann** [Zipios/Zipios#43]
Hi,

the zip file format allows some variants. One of them is "trailing data descriptors", see APPNOTE.TXT of 2020, section 4.4.4, Bit 3:
[https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT)
To quote Wikipedia:
"If the bit at offset 3 (0x08) of the general-purpose flags field is set, then the CRC-32 and file sizes are not known when the header is written. If the archive is in Zip64 format, the compressed and uncompressed size fields are 8 bytes long instead of 4 bytes long [...]. The equivalent fields in the local header (or in the Zip64 extended information extra field in the case of archives in Zip64 format) are filled with zero, and the CRC-32 and size are appended in a 12-byte structure (optionally preceded by a 4-byte signature) immediately after the compressed data [...]"

Zipios never officially supported this variant. The current situation is that on identification of a zip file with trailing data descriptor, an exception is thrown: zipinputstreambuf.cpp, line 66.

Pre-1.0-versions of zipios didn't support trailing data descriptors either, but due to a simple bug zipios 0.1.* didn't identify trailing data descriptors as such and deflated them anyway (at least when the archive contains only one file).

Trailing data descriptors are used when the exact sizes of files are not known beforehand, which might be quite a use case for input-output-streams. If zipios is to support the zip file format, it should support trailing data descriptors in compressing and decompressing as well, I guess?

Please find attached an example zip.
[trailing.zip](https://github.com/Zipios/Zipios/files/10937206/trailing.zip)

Best,
Andreas

**Comment by AlexisWilke** [comment]
Yeah, that would probably be a good idea. I did not realize that caused an issue.

**Comment by luzpaz** [comment]
Any traction here ?

**Comment by luzpaz** [comment]
Softest of bumps  
(Note: FreeCAD has hacked a workaround in the meantime, see https://github.com/FreeCAD/FreeCAD/pull/15988)
